assets/js/push.js

```diff
@@ -122,6 +148,45 @@ jQuery( window ).on( 'load', () => {
 	function distributorMenuEntered() {
 		distributorMenuItem.focus();
 		document.body.classList.toggle( 'distributor-show' );
+
+		if ( distributorPushWrapper.classList.contains( 'loaded' ) ) {
+			return;
+		}
+
+		distributorPushWrapper.classList.remove( 'message-error' );
+		distributorPushWrapper.classList.add( 'loaded' );
+
+		const data = {
+			action: 'dt_load_connections',
+			loadConnectionsNonce: dt.loadConnectionsNonce,
+			postId: dt.postId
+		};
+
+		const xhr = dt.usexhr ? { withCredentials: true } : false;
+
+		jQuery.ajax( {
+			url: dt.ajaxurl,
+			xhrFields: xhr,
+			method: 'post',
+			data: data
+		} ).done( ( response ) => {
+			if ( ! response.success || ! response.data ) {
+				distributorPushWrapper.classList.remove( 'loaded' );
+				distributorPushWrapper.classList.add( 'message-error' );
+				return;
+			}
+
+			dtConnections = response.data;
+
+			distributorPushWrapper.innerHTML = processTemplate( 'dt-show-connections' )( {
```

Comment:
@dkotter unfortunately `innerHTML` is insecure, if a malicious attacker were able to hijack the ajax request, they could return a malicious script that would get injected here. 

Ideal fix here would be to return a data structure and then construct the HTML using DOM construction or a template structure (template literals would work as well). Another option would be accepting the risk - it certainly is easier/feels safer leaving the HTML generation in place.

Reply:
Okay. Currently we have a javascript template we output within `push-ui.php` (line 437), which we then take that and use underscores JS to take that template, plus the data from our ajax request, to build out the markup (which we then add to the element using `innerHTML`). Note the data we use in this template is escaped using underscores escaping function (example `{{ connections[ key ]['name'] }}`.

So we do currently use a template structure here but are you saying that should be done differently? Note I copied what we were already doing another place in this same file (line 141), just to keep things consistent. Doesn't mean that's the right approach but if we want to change this, we'll probably want to change in both places.